joomla/joomla-cms
# [3.x] com_wrapper. Add HTML5 mode without B\C break.

Related to https://github.com/joomla/joomla-cms/pull/19965
See discussion there concerning B\C breaks in Joomla 3.x.

@BaleshSrle
**Please feel free to take over code and/or idea into your pr. I have no problems to close this one then.**

### Summary of Changes
- Adds a new switch `Legacy (HTML4)` versus `HTML5`.
- `Legacy` is the default setting. Thus, this pr will not break older template overrides that use HTML4 attributes for IFRAME tags that have been removed in HTML5.
- Doesn't touch old parameter types and names but translates them via PHP for `HTML5` mode if necessary.
- `HTML5` uses CSS style instead. 
- Comment: Because most current browsers don't support CSS rule `overflow` for displaying/hiding scroll bars `scrolling` attribute cannot be removed in HTML5 mode.

### Testing Instructions
- Apply patch.
- Create a menu item of type "Wrapper" with setting `Mode = Legacy`.
- Create another menu item of type "Wrapper" with setting `Mode = HTML5`.

![25-03-_2018_15-55-15](https://user-images.githubusercontent.com/20780646/37875855-0bd91812-3045-11e8-9025-d64a60ff9121.png)

- Compare outputs in frontend.
- Test other settings and compare output in frontend.
- As far as I see: Old template overrides should still work even if you switch mode to `HTML5` because only default.php of core respects new setting. 

### Expected result
- No optical difference
 
### Documentation Changes Required
????

### New language strings
YES
Please help to optimize description.
